site/functions/server/main.go

```diff
@@ -0,0 +1,287 @@
+	var out []*github.RepositoryContent
+	for _, v := range entries {
+		if v == nil {
+			continue
+		}
+		if v.GetType() == "file" && strings.HasSuffix(v.GetName(), constants.ManifestExtension) {
+			out = append(out, v)
+		}
+	}
+	return out
+}
+
+func loggingHandler(f http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		log.Printf("[req]  > method=%s path=%s", req.Method, req.URL)
+		defer func() {
+			log.Printf("[resp] < method=%s path=%s took=%v", req.Method, req.URL, time.Since(start))
+		}()
+		f.ServeHTTP(w, req)
+	})
+}
+
+func pluginsHandler(w http.ResponseWriter, req *http.Request) {
+	_, dir, resp, err := githubClient(req.Context()).
+		Repositories.GetContents(req.Context(), orgName, repoName, pluginsDir, &github.RepositoryContentGetOptions{})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("error retrieving repo contents: %v", err)}})
+		return
+	}
+	log.Printf("github response=%s rate: limit=%d remaining=%d",
+		resp.Status, resp.Rate.Limit, resp.Rate.Remaining)
+	var out PluginsResponse
+
+	plugins, err := fetchPlugins(filterYAMLs(dir))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginsResponse{Error: ErrorResponse{Message: fmt.Sprintf("failed to fetch plugins: %v", err)}})
+		return
+	}
+
+	for _, v := range plugins {
+		pi := pluginInfo{
+			Name:             v.Name,
+			Homepage:         v.Spec.Homepage,
+			ShortDescription: v.Spec.ShortDescription,
+			GithubRepo:       findRepo(v.Spec.Homepage),
+		}
+		out.Data.Plugins = append(out.Data.Plugins, pi)
+	}
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func fetchPlugins(entries []*github.RepositoryContent) ([]*krew.Plugin, error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var (
+		mu     sync.Mutex
+		out    []*krew.Plugin
+		retErr error
+	)
+
+	queue := make(chan string)
+	var wg sync.WaitGroup
+
+	for i := 0; i < urlFetchBatchSize; i++ {
+		wg.Add(1)
+		go func(j int) {
+			defer wg.Done()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case url, ok := <-queue:
+					if !ok {
+						return
+					}
+					p, err := readPlugin(url)
+					if err != nil {
+						retErr = err
+						cancel()
+						return
+					}
+					mu.Lock()
+					out = append(out, p)
+					mu.Unlock()
+				}
+			}
+		}(i)
+	}
+
+	for _, v := range entries {
+		url := v.GetDownloadURL()
+		select {
+		case <-ctx.Done():
+			break
+		case queue <- url:
+		}
+	}
+
+	close(queue)
+	wg.Wait()
+
+	return out, retErr
+}
+
+func readPlugin(url string) (*krew.Plugin, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err,"failed to get %s", url)
+	}
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
```

Comment:
nit:
```suggestion
	defer resp.Body.Close()
```

From the spec:
> // The http Client and Transport guarantee that Body is always
	// non-nil, even on responses without a body or responses with
	// a zero-length body. It is the caller's responsibility to
	// close Body.